packages/vanilla/src/components/masthead/mastheadHeaderMenuButton.template.js

```diff
@@ -0,0 +1,62 @@
+import { settings } from 'carbon-components';
+import { toString } from '@carbon/icon-helpers';
+import { Close20, Menu20 } from '@carbon/icons';
+import cx from 'classnames';
+
+const { prefix } = settings;
+
+/**
+ * Masthead Header Button Menu
+ *
+ * @param {object} props Property object passed into the template
+ * @param {string} props.className Custom css classes passed into the root element
+ * @param {object} props.attrs Object with key/value HTML attributes passed into the root element
+ * @param {boolean} props.isActive Boolean to show proper content based on active state
+ * @param {boolean} props.isCollapsible Boolean to show proper content based on collapsible state
+ * @returns {string} ES6 template literal of the horizontal rule
+ *
+ */
+const mastheadHeaderMenuButtonTemplate = (props = {}) => {
+  const {
+    className: customClassName,
+    attrs = {},
+    isActive,
+    isCollapsible,
+  } = props;
+  const { 'aria-label': ariaLabel } = attrs;
+
+  // TODO: Check if should use root
+  window.onHeaderMenuButtonClick = () => {
```

Comment:
Should use `window-or-root` to help determine. Should assume that adopters would be using this either client side or server side.